src/format.rs

```diff
@@ -187,19 +187,57 @@ impl PayloadParams {
         let c1 = o.spec;
 
         if c0 == c1 {
+            // Exact match
             return Some(100);
-        } else {
-            // TODO: fuzzy matching.
         }
 
-        // No match
+        // Attempt fuzzy matching, since we don't have an exact match
+        if c0.codec != c1.codec || (c0.codec == Codec::Unknown && c1.codec == Codec::Unknown) {
+            // Codecs must match
+            return None;
+        }
+
+        if c0.clock_rate != c1.clock_rate {
+            // Clock rates must match
+            return None;
+        }
+
+        if c0.channels != c1.channels {
+            // Channels must match
+            return None;
+        }
+
+        if c0.codec.is_audio() {
+            let mut score: usize = 100;
+            // If neither value is specified both sides should assume the default value, 3.
+            let either_p_time_specified =
```

Comment:
Wonder if we should break this out into a `match_score_opus`, since the logic here I think it opus specific.

Reply:
I was thinking about splitting this like that, but I kept it here for clarity since the amount of code is limited. If we start supporting format parameters for other audio codecs we should probably make a function.

Related point: should we have an if-guard to check the codec before this code?